chain-impl-mockchain/src/certificate/mint_token.rs

```diff
@@ -0,0 +1,101 @@
+use crate::{
+    account::Identifier,
+    certificate::CertificateSlice,
+    tokens::{MintingPolicy, TokenIdentifier},
+    transaction::{Payload, PayloadAuthData, PayloadData, PayloadSlice},
+    value::Value,
+};
+
+use chain_core::mempack::{ReadBuf, ReadError, Readable};
+use typed_bytes::ByteBuilder;
+
+use std::marker::PhantomData;
+
+#[derive(Debug, Clone)]
+pub struct MintToken {
+    pub token: TokenIdentifier,
```

Comment:
What about making the fields private to enforce consistency between token and minting policy?
Also keeping the hash of the policy in `token` is a bit redundant at this point. Not storing a copy of the policy hash would solve both cases

Reply:
> Also keeping the hash of the policy in token is a bit redundant at this point.

It is a bit redundant _at this point_ but this will change in near future.

Reply:
> What about making the fields private to enforce consistency between token and minting policy?

How would it help?

Reply:
> It is a bit redundant at this point but this will change in near future.

What do you mean? Isn't the policy hash always derived from hashing data you already keep in the struct?

> What about making the fields private to enforce consistency between token and minting policy?
How would it help?

I  cannot write something like
```
MintToken { token: TokenIdentifier::new("token", TestGen::hash()), policy: somepolicy}
```

Reply:
> I cannot write something like
>
>     MintToken { token: TokenIdentifier::new("token", TestGen::hash()), policy: somepolicy}

Ah, got it, makes sense.

> What do you mean? Isn't the policy hash always derived from hashing data you already keep in the struct?

Wait, so just make it clear. You mean that in the `MintToken` certificate it doesn't make sense to keep the policy hash because the policy itself is already there? If so, it was a misunderstanding and you're right, the policy hash as a certificate field is redundant.

Reply:
> You mean that in the MintToken certificate it doesn't make sense to keep the policy hash because the policy itself is already there? If so, it was a misunderstanding and you're right, the policy hash as a certificate field is redundant.

yep, exactly